.github/workflows/ci.yml

```diff
@@ -3,15 +3,47 @@ name: ci
 on: [push, pull_request]
 
 env:
-  X_PYTHON_VERSION: "3.12"
+  X_PYTHON_MIN_VERSION: "3.9"
+  X_PYTHON_MAX_VERSION: "3.12"
```

Comment:
I think it would be good to define a list of supported Python versions as env variable here and use it for matrices.